### PR TITLE
Feature/issue 373

### PR DESCRIPTION
**Pull request by jbiggsets**
This is a PR to address #373. It updates all of the docstrings in the `skll` package. There are two main types of changes:

1. Converting **existing** docstrings to `numpy`-style. This sometimes involved adding additional information, such as any errors being raised or arguments the were not mentioned.
2. Adding **new** `numpy`-style docstrings to functions and methods where none previously existed.

I have also added the `sphinx.ext.napoleon` extension to the `doc/config.py` file to properly parse `numpy`-style docstrings when generating documentation for RTD.

Please review _new_ docstrings especially carefully, with an eye toward proper argument and return data types.

### CODE REVIEW

**Comment by jbiggsets** [comment]
This is ready for review. @desilinguist, I'm no longer getting any warnings when I generate the documentation with `make clean && make html` locally.

**Comment by desilinguist** [comment]
Okay, now that the coverage issue is resolved, I am going to start to review this.

File: skll/config.py [hunk 1]
```diff
@@ -141,7 +143,12 @@ def __init__(self):
 
     def _find_invalid_options(self):
         """
-        Returns the set of invalid options specified by the user
+        Find the set of invalid options specified by the user.
+
+        Returns
+        -------
+        invalid_options : set
+            The set of invalid options specified by the user.
```
Comment (Lguyogiro) [inline skll/config.py]: Do we want to specify the type of the set elements to expect? eg `set of str`. This change would apply to a lot of other functions as well.

**Comment by desilinguist** [comment]
@aoifecahill Once the build passes, this is just waiting for your approval.

**Pull request merged**